Params inputParams
Scenario scenario
Vehicle vehicle